src/msix/PAL/XML/xerces-c/XmlObject.cpp

```diff
@@ -402,6 +533,124 @@ class XercesDom final : public ComClass<XercesDom, IXmlDom>
     }
 
 protected:
+
+    std::unique_ptr<MemBufInputSource> StripIgnorableNamespaces(const InputSource& source, const NamespaceManager& namespaces)
+    {
+        m_parser->setDoNamespaces(true);
+        m_parser->parse(source);
+        XERCES_CPP_NAMESPACE::DOMDocument* dom = m_parser->getDocument();
+        auto rootElement = ComPtr<IXercesElement>::Make<XercesElement>(m_factory, dom->getDocumentElement(), m_parser.get());
+        std::string attr = "IgnorableNamespaces";
+        std::string attrValue = rootElement->GetAttributeValue(attr);
+        if (!attrValue.empty())
+        {
+            std::vector<std::string> aliasesToLookup;
+            {
+                std::string alias;
+                std::istringstream aliases(attrValue);
+                while(getline(aliases, alias, ' ')) { aliasesToLookup.push_back(alias); }
+            }
+            for (const auto& a : aliasesToLookup)
+            {
+                std::string alias = "xmlns:" + a; // Look for xmlns:[alias] attribute name
+                std::string aliasValue = rootElement->GetAttributeValue(alias);
+                const auto& entry = std::find(namespaces.begin(), namespaces.end(), aliasValue.c_str());
+                if (entry == namespaces.end()) // only strip if we don't know about it
+                {
+                    RemoveAllInNamespace(rootElement, a);
+                }
+            }
+
+        }
+
+        // Serialize the new dom to parse.
+        static const XMLCh cs[3] = {chLatin_L, chLatin_S, chNull};
+        DOMImplementation *impl = DOMImplementationRegistry::getDOMImplementation(cs);
+        auto serializer = XercesPtr<DOMLSSerializer>((static_cast<DOMImplementationLS*>(impl))->createLSSerializer());
+        auto lsOutput = XercesPtr<DOMLSOutput>((static_cast<DOMImplementationLS*>(impl))->createLSOutput());
+
+        // Set encoding to UTF-8
+        static const XMLCh utf8Str[] = {chLatin_U, chLatin_T, chLatin_F, chDash, chDigit_8, chNull};
+        lsOutput->setEncoding(utf8Str);
+
+        // Don't use unique_ptr here, this buffer is going to be deleted by Xerces

```

Comment:
>Don't use unique_ptr here, this buffer is going to be deleted by Xerces [](start = 11, length = 71)

Comment is confusing since the very next thing you do is use a unique_ptr

Reply:
yeah, this comment was supposed to be before copying the buffer.  moved

---
In reply to: [273624046](https://github.com/Microsoft/msix-packaging/pull/82#discussion_r273624046) [](ancestors = 273624046)